spec/models/request/sample_compound_aliquot_transfer_spec.rb

```diff
@@ -75,15 +75,98 @@
              library_id: 54
     end
 
-    it 'creates a compound sample and transfers an aliquot of it' do
-      expect(sequencing_request.target_asset.aliquots.count).to eq(0)
-      sequencing_request.transfer_aliquots_into_compound_sample_aliquots
-      expect(sequencing_request.target_asset.aliquots.count).to eq(1)
-      expect(sequencing_request.target_asset.aliquots.first.library_type).to eq('Standard')
-      expect(sequencing_request.target_asset.aliquots.first.study).to eq(study1)
-      expect(sequencing_request.target_asset.aliquots.first.project).to eq(project1)
-      expect(sequencing_request.target_asset.aliquots.first.library_id).to eq(54)
-      expect(sequencing_request.target_asset.samples.first.component_samples.order(:id)).to eq(samples.sort)
+    context 'when no compound sample exists with the component samples' do
+      it 'creates a compound sample and transfers an aliquot to it' do
+        expect(sequencing_request.target_asset.aliquots.count).to eq(0)
+
+        expect { sequencing_request.transfer_aliquots_into_compound_sample_aliquots }.to change(Sample, :count).by(1)
+
+        expect(sequencing_request.target_asset.aliquots.count).to eq(1)
+        expect(sequencing_request.target_asset.aliquots.first.library_type).to eq('Standard')
+        expect(sequencing_request.target_asset.aliquots.first.study).to eq(study1)
+        expect(sequencing_request.target_asset.aliquots.first.project).to eq(project1)
+        expect(sequencing_request.target_asset.aliquots.first.library_id).to eq(54)
+        expect(sequencing_request.target_asset.samples.first.component_samples.count).to eq 2
+        expect(sequencing_request.target_asset.samples.first.component_samples[0]).to eq samples[0]
+        expect(sequencing_request.target_asset.samples.first.component_samples[1]).to eq samples[1]
+      end
+    end
+
+    context 'when a compound sample exists with different component samples' do
+      let(:compound_sample) { create(:sample, name: 'compound_sample_1') }
+
+      before do
+        compound_sample.update(component_samples: [samples[0], samples[2]])
```

Comment:
Its generally better to pass the samples in to the factory, than mutate it later.

1) It makes it easier to see at a glance the state of the compound_sample, as you only need to look at the let.
2) You won't be caught out if the factory is also creating samples
3) In many cases it is faster, as it reduces the trips to the database - not sure if that will be true in this context though.

In this case you don't actually refer to compound sample outside of the before block, so you can just make the call to the factory there, rather than having a named let block.